### PR TITLE
Slider

### PR DESCRIPTION
**Pull request by dtorrenp**
#10 

Created a slider based on the html input range element, with the ability to show line and number ticks. A different styling fro when it is in "play" mode which includes extremely basic previous and next buttons. The slider ticks re-size when the screen width is below a given size (600px).

slider_basic

### CODE REVIEW

**Comment by dtorrenp** [comment]
I took out the console log statements and resolved the merge issues

**Comment by darrenlean** [comment]
Hey Dan,

You might need a break statement to prevent 0 tick and someway to handle overcrowded tick labels:

Screenshot 2020-07-07 at 3 43 27 pm

File: package.json [hunk 1]
```diff
@@ -17,7 +17,14 @@
...
     "katex": "^0.11.1",
-    "save": "^2.4.0"
+    "save": "^2.4.0",
+    "vue": "^2.6.11",
+    "vue-katex": "^0.4.0",
+    "vue-katex-auto-render": "^0.1.2",
+    "vue-multiselect": "^2.1.6",
+    "vue-plotly": "^1.1.0",
+    "vue-resize": "^0.5.0",
+    "vue-resize-sensor": "^2.0.0"
```
Comment (OneOneFour) [inline package.json:27]: remove vue-katex,vue-katex-auto-render and vue-plotly and vue-multiselect

File: src/components/Slider/Slider.vue [hunk 2]
```diff
@@ -0,0 +1,328 @@
+<template>
+    <div class = "sliderContainer" ref="sliderContainer">
+        <resize-observer @notify="update_step" />
+        <div class = "titleGroup">
+            <p v-if="description_val">{{description}}</p>
+            <p v-if="sliderValue">{{value}}</p>
+        </div>
```
Comment (OneOneFour) [inline src/components/Slider/Slider.vue:7]: I would remove this block here, as this is a bit of an over-engineer

File: src/components/Slider/Slider.vue [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        },
+        playSlider:{
+            type:Boolean,
+            default:false
+        },
+        description_val:{
+            type:Boolean,
+            default:false
+        },
+        description:{
```
Comment (OneOneFour) [inline src/components/Slider/Slider.vue:82]: Remove this prop if the block as whole is being removed

File: src/components/Slider/Slider.vue [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+                tick_list.push({id: this.id.toString() + "_" + i.toString(), value: i.toFixed(this.tick_decimals)});
+            }
+            return tick_list
+        },
+        emitSlider(){
+            this.$emit("sliderChanged",this.value);
+        }
+    },
+    mounted () {
+        this.id = this._uid;
```
Comment (OneOneFour) [inline src/components/Slider/Slider.vue:203]: Nice 👍

File: src/components/Slider/Slider.vue [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+.iv-range-play::-webkit-slider-runnable-track{
+    -webkit-appearance: none;
+    border: 1px solid black;
+    border-radius: 9px;
+    background-color:cornsilk;
+}
+
+/* thumb css (dont know if this needs to be specified)*/
+.iv-range::-webkit-slider-thumb{
```
Comment (OneOneFour) [inline src/components/Slider/Slider.vue:278]: You are using webkit-slider-thumb here, have you checked if this component works in other browsers:
would check edge and firefox to see if that also works?

File: src/components/Slider/bubble-comp.vue [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<template>
```
Comment (OneOneFour) [inline src/components/Slider/bubble-comp.vue:1]: Change the name of this file to Bubble.vue

File: src/components/Slider/num-ticks.vue [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<template>
```
Comment (OneOneFour) [inline src/components/Slider/num-ticks.vue:1]: As a style thing i would also write filenames in camelCase one word no hyphen

File: package-lock.json [hunk 8]
```diff
@@ -12062,6 +12129,8 @@
         "vue-style-loader": "^4.1.0"
       },
       "dependencies": {
+<<<<<<< HEAD
```
Comment (OneOneFour) [inline package-lock.json:12132]: There seems to be an unhandled git merge in your package-lock.json.
Either resolve it, or just remove this file your PR

File: package-lock.json [hunk 9]
```diff
@@ -12086,9 +12176,95 @@
...
+          "dev": true,
+          "requires": {
+            "ansi-styles": "^2.2.1",
+            "escape-string-regexp": "^1.0.2",
+            "has-ansi": "^2.0.0",
+            "strip-ansi": "^3.0.0",
+            "supports-color": "^2.0.0"
+          }
+        },
+>>>>>>> a685ff5a4be8105c16db117d916aeffd27f73106
```
Comment (OneOneFour) [inline package-lock.json:12245]: Git merge conflict ends here

**Comment by dtorrenp** [comment]
so i tried deleting the package files, pulling and then pushing again, tbh im a little confused as to why this merge error keeps happening.

**Comment by OneOneFour** [comment]
Huh, seems like the linter requires a package.json to be present

**Comment by OneOneFour** [comment]
I think you should probably reset the package.json to match the exact version of package.json (you don't need a package-lock tho) that is on upstream/dev

**Pull request merged**